addons/html_editor/static/src/main/hint_plugin.js

```diff
@@ -72,18 +72,50 @@ export class HintPlugin extends Plugin {
     updateHints() {
         const selectionData = this.shared.getSelectionData();
         const editableSelection = selectionData.editableSelection;
+        const blockEl = closestBlock(editableSelection.anchorNode);
         if (this.hint) {
-            const blockEl = closestBlock(editableSelection.anchorNode);
             this.removeHint(this.hint);
             this.removeHint(blockEl);
         }
         if (editableSelection.isCollapsed) {
+            const columnContainer = closestElement(editableSelection.anchorNode, ".o_text_columns");
+            if (columnContainer) {
+                this.hint = [];
+            }
             for (const hint of this.getResource("hints")) {
                 if (hint.selector) {
-                    const el = closestBlock(editableSelection.anchorNode);
-                    if (el && el.matches(hint.selector) && !isProtected(el) && isEmptyBlock(el)) {
-                        this.makeHint(el, hint.text);
-                        this.hint = el;
+                    if (columnContainer) {
+                        for (const elem of selectElements(
+                            columnContainer,
+                            `${hint.selector}:first-child`
+                        )) {
+                            // Apply helper hint on first block element of each column
+                            if (
+                                elem.closest("div[class^='col-']") ===
+                                    closestElement(
+                                        editableSelection.anchorNode,
+                                        "div[class^='col-']"
+                                    ) &&
+                                blockEl !== elem
+                            ) {
+                                // If cursor is not inside first block element but a different
+                                // one, skip that first element.
+                                continue;
+                            }
+                            if (!isProtected(elem) && isEmptyBlock(elem)) {
+                                this.makeHint(elem, hint.text);
+                                this.hint.push(elem);
+                            }
+                        }
+                    }
+                    if (
+                        blockEl &&
+                        blockEl.matches(hint.selector) &&
+                        !isProtected(blockEl) &&
+                        isEmptyBlock(blockEl)
+                    ) {
+                        this.makeHint(blockEl, hint.text);
+                        Array.isArray(this.hint) ? this.hint.push(blockEl) : (this.hint = blockEl);
```

Comment:
Here I think the code is not at the correct place. The logic for columns hint should note be in the hintPlugin.
How about you use the target mechanism and replace the selector coming from `columnPlugin` to target. Something similar to what is done in `hintPlugin` itself.